pkg/security/ebpf/c/include/hooks/network/accept.h

```diff
@@ -0,0 +1,69 @@
+#ifndef _HOOKS_ACCEPT_H_
+#define _HOOKS_ACCEPT_H_
+
+#include "constants/offsets/network.h"
+
+int __attribute__((always_inline)) read_sock_and_send_event(ctx_t * ctx, struct sock * sock) {
+    struct accept_event_t event = {0};
+
+    // Extract family from the socket
+    struct sock_common *sockcommon = (void *)sock;
+    event.family = get_family_from_sock_common(sockcommon);
+    // Only handle AF_INET and AF_INET6
+    if (event.family != AF_INET && event.family != AF_INET6) {
+        return 0;
+    }
+
+    // Read the listening port and source address
+    if (event.family == AF_INET) {
+        bpf_probe_read(&event.port, sizeof(event.port), &sockcommon->skc_num);
+        bpf_probe_read(&event.addr[0], sizeof(event.addr[0]), &sockcommon->skc_daddr);
+    } else if (event.family == AF_INET6) {
+        bpf_probe_read(&event.port, sizeof(event.port), &sockcommon->skc_num);
+        bpf_probe_read((void*)&event.addr, sizeof(sockcommon->skc_v6_daddr), &sockcommon->skc_v6_daddr);
+    }
+
+    event.port = htons(event.port);
+
+    struct proc_cache_t *entry = fill_process_context(&event.process);
+    fill_container_context(entry, &event.container);
+    fill_span_context(&event.span);
+
+    // Check if we should sample this event for activity dumps
+    struct activity_dump_config *config = lookup_or_delete_traced_pid(event.process.pid, bpf_ktime_get_ns(), NULL);
```

Comment:
as we don't add accept in activity dump yet, you could remove this section

Reply:
Addressed